addons/binding/org.openhab.binding.squeezebox/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,243 @@
+		<label>SqueezeBox Server</label>
+		<description>This is a SqueezeBox Server instance.
+		</description>
+
+		<config-description>
+			<parameter name="ipAddress" type="text" required="true">
+				<context>network-address</context>
+				<label>IP or Host Name</label>
+				<description>The IP or host name of the SqueezeServer
+				</description>
+			</parameter>
+			<parameter name="webport" type="integer" required="true"
+				min="1" max="65335">
+				<context>network-address</context>
+				<label>SqueezeServer Web Port</label>
+				<description>Webport interface of the SqueezeServer</description>
+				<default>9000</default>
+			</parameter>
+			<parameter name="cliport" type="integer" required="true"
+				min="1" max="65335">
+				<context>network-address</context>
+				<label>SqueezeServer CLI Port</label>
+				<description>Port of the CLI interface of the SqueezeServer</description>
+				<default>9090</default>
+			</parameter>
+			<parameter name="language" type="text" required="false">
+				<label>Language</label>
+				<description>Language to use when using Google speech</description>
+				<default>en</default>
+			</parameter>
+		</config-description>
+	</bridge-type>
+
+	<thing-type id="squeezeboxplayer">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="squeezeboxserver" />
+		</supported-bridge-type-refs>
+		<label>SqueezeBox Player</label>
+		<description>This is a SqueezeBox Player instance</description>
+		<channels>
+			<channel id="power" typeId="power" />
+			<channel id="mute" typeId="mute" />
+			<channel id="volume" typeId="volume" />
+			<channel id="stop" typeId="stop" />
+			<channel id="control" typeId="control" />
+			<channel id="stream" typeId="stream" />
+			<channel id="sync" typeId="sync" />
+			<channel id="playListIndex" typeId="playListIndex" />
+			<channel id="currentPlayingTime" typeId="currentPlayingTime" />
+			<channel id="currentPlaylistShuffle" typeId="currentPlaylistShuffle" />
+			<channel id="currentPlaylistRepeat" typeId="currentPlaylistRepeat" />
+			<channel id="title" typeId="title" />
+			<channel id="remotetitle" typeId="remotetitle" />
+			<channel id="album" typeId="album" />
+			<channel id="artist" typeId="artist" />
+			<channel id="year" typeId="year" />
+			<channel id="genre" typeId="genre" />
+			<channel id="coverarturl" typeId="coverarturl" />
+			<channel id="coverartdata" typeId="coverartdata" />
+			<channel id="ircode" typeId="ircode" />
+			<channel id="numberPlaylistTracks" typeId="numberPlaylistTracks" />
+			<channel id="ip" typeId="ip" />
+			<channel id="uid" typeId="uid" />
+			<channel id="name" typeId="name" />
+			<channel id="model" typeId="model" />
+		</channels>
+		<config-description>
+			<parameter name="mac" type="text" required="true">
+				<label>MAC Address</label>
+				<description>SqueezeBox Players are identified by their MAC address</description>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<!-- Commands -->
```

Comment:
you might want to mark many channel types as "advanced", so that only the most important ones (power, control, volume,...) are directly shown to the user.